src/vmm/src/vstate/vcpu/x86_64.rs

```diff
@@ -348,6 +393,9 @@ impl KvmVcpu {
 
         Self::fix_zero_tsc_deadline_msr(&mut msr_chunks);
 
+        let deferred = Self::defer_msrs(&mut msr_chunks)?;
+        msr_chunks.push(deferred);
```

Comment:
The name `defer_msrs` in my mind sounds like we are differing them in the method, but we actually just remove them. So maybe either rename `defer_msrs` to `remove_defer_msrs` or move `msr_chunks.push(deferred);` into the `defer_msrs` as well?

Reply:
Good point, originally I wanted to have the `push` inside the function, but couldn't because of it only taking a slice. But then I never updated the function name to reflect what is actually going on. Now its called `extract_deferred_msrs`.